data/events/alerts.go

```diff
@@ -0,0 +1,354 @@
+package events
+
+import (
+	"cmp"
+	"context"
+	"os"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/IBM/sarama"
+	"go.mongodb.org/mongo-driver/bson"
+
+	"github.com/tidepool-org/platform/alerts"
+	"github.com/tidepool-org/platform/auth"
+	"github.com/tidepool-org/platform/data/store"
+	"github.com/tidepool-org/platform/data/types/blood/glucose"
+	"github.com/tidepool-org/platform/data/types/dosingdecision"
+	"github.com/tidepool-org/platform/devicetokens"
+	"github.com/tidepool-org/platform/errors"
+	"github.com/tidepool-org/platform/log"
+	logjson "github.com/tidepool-org/platform/log/json"
+	lognull "github.com/tidepool-org/platform/log/null"
+	"github.com/tidepool-org/platform/permission"
+	"github.com/tidepool-org/platform/push"
+)
+
+type Consumer struct {
+	Alerts       AlertsClient
+	Data         store.DataRepository
+	DeviceTokens auth.DeviceTokensClient
+	Evaluator    AlertsEvaluator
+	Permissions  permission.Client
+	Pusher       Pusher
+	Tokens       alerts.TokenProvider
+
+	Logger log.Logger
+}
+
+// DosingDecision removes a stutter to improve readability.
+type DosingDecision = dosingdecision.DosingDecision
+
+// Glucose removes a stutter to improve readability.
+type Glucose = glucose.Glucose
+
+func (c *Consumer) Consume(ctx context.Context,
+	session sarama.ConsumerGroupSession, msg *sarama.ConsumerMessage) (err error) {
+
+	if msg == nil {
+		c.logger(ctx).Info("UNEXPECTED: nil message; ignoring")
+		return nil
+	}
+
+	switch {
+	case strings.HasSuffix(msg.Topic, ".data.alerts"):
+		return c.consumeAlertsConfigs(ctx, session, msg)
+	case strings.HasSuffix(msg.Topic, ".data.deviceData.alerts"):
+		return c.consumeDeviceData(ctx, session, msg)
+	default:
+		c.logger(ctx).WithField("topic", msg.Topic).
+			Infof("UNEXPECTED: topic; ignoring")
+	}
+
+	return nil
+}
+
+func (c *Consumer) consumeAlertsConfigs(ctx context.Context,
+	session sarama.ConsumerGroupSession, msg *sarama.ConsumerMessage) error {
+
+	cfg := &alerts.Config{}
+	if err := unmarshalMessageValue(msg.Value, cfg); err != nil {
+		return err
+	}
+	lgr := c.logger(ctx)
+	lgr.WithField("cfg", cfg).Info("consuming an alerts config message")
+
+	ctxLog := c.logger(ctx).WithField("followedUserID", cfg.FollowedUserID)
+	ctx = log.NewContextWithLogger(ctx, ctxLog)
+
+	notes, err := c.Evaluator.Evaluate(ctx, cfg.FollowedUserID)
```

Comment:
In that case, there's a repetition going on, because the package's name is "alerts", so you end up with "alerts.AlertNotification". Would you be ok with "Notification" for the type, and a change for variable names from "note" => "notification"?

If you're editing a file within the alerts package, it should be clear that a Notification is a notification of an alert, no?

Reply:
> Would you be ok with "Notification" for the type, and a change for variable names from "note" => "notification"?

Sure, that sounds reasonable